### DISCUSSION TITLE
Probat P12III , connection problems

### DISCUSSION THREAD

**Discussion by MatteoNat** [artisan-roaster-scope/artisan#1525] [Q&A] [answered]
Hi everyone, I'm Matteo
I'm writing here to try to solve a connection problem.
My laptop is connected directly with an Ethernet cable to our Probat P12III series.
I already tried creating a static IP address. I disabled the firewalls and uninstalled the antivirus.
We were in Team Viewer with a Probat ing software. staff who tried to solve my problem but we couldn't.
On the first day everything worked and now Artisan does not show any readings. The cable is well connected as, using the IP and the Google page, I can have the control panel projected on my laptop.
Any idea what could be wrong?

Many thanks to everyone.

Matteo
![Artisan graph](https://github.com/artisan-roaster-scope/artisan/assets/165190223/7153567d-18f5-4af5-8bc2-9545da440494)

**Comment by MAKOMO** [comment]
> Hi everyone, I'm Matteo I'm writing here to try to solve a connection problem. My laptop is connected directly with an Ethernet cable to our Probat P12III series. I already tried creating a static IP address. 

Why?

> I disabled the firewalls and uninstalled the antivirus. 

Why?

> We were in Team Viewer with a Probat ing software. staff who tried to solve my problem but we couldn't. 

Where you talking with people from Probat HQ or others?

What did they try?

>On the first day everything worked and now Artisan does not show any readings. 

What did you change between the working state and the non-working state?

>The cable is well connected as, 

Which cable? The one of the roaster?

> using the IP and the Google page, I can have the control panel projected on my laptop. 

I cannot follow you here. Which IP and Google page? Which "control panel" do you get "projected" on your laptop? Maybe you can connect to the roaster via the browser on your laptop? Which URL address are you using for this in your Web browser? Could you please paste it here?

> Any idea what could be wrong?

Your report lacks a lot information. From the screenshot I can see that you are on Windows using Artisan v2.10.2.

Could you attach a picture of your roasters control panel showing its IP address?
Could you please attach screenshots of your Windows network settings?
Could you please attach a screenshot of your Artisans WebSocket tab (menu Config >> Port, last tab "WebSocket"?


> 
> Many thanks to everyone.
> 
> Matteo

**Reply by MatteoNat** [reply]
Going by steps 
1) As I'm using the mobile phone as a internet connection they told me the ip can change and that may be the problem.
2)To avoid any firewall or antivirus system to clash with the software 
3) HQ Probat service: they tried static IP, unistall and reinstall, they went manually in port, machine and ip configurations. 
4) No changes
5) An Ethernet cable directly connected from the laptop to the roaster (back machine Ethernet plug)
6) I'm just writing machine ip address in the link bar to have this 
![Probat proj](https://github.com/artisan-roaster-scope/artisan/assets/165190223/8e57ce1f-ca9d-42b4-a2c1-80b91c65f25a)
![probat proj 2](https://github.com/artisan-roaster-scope/artisan/assets/165190223/ae0783e3-c1a8-49e9-a1ac-85faf106048c)
7) I'm using Artisan 2.10.4
![Artisan graph](https://github.com/artisan-roaster-scope/artisan/assets/165190223/8bb71b04-cf5f-4906-a290-b6e27413e600)
![20240328_151734](https://github.com/artisan-roaster-scope/artisan/assets/165190223/78d545e5-e0c1-49a2-8ad5-3405ecd34b96)
![Int set](https://github.com/artisan-roaster-scope/artisan/assets/165190223/f71d013a-2bac-4082-8393-936aa320c5f9)
![Webs](https://github.com/artisan-roaster-scope/artisan/assets/165190223/f7b7b447-7756-4e58-b4b2-957ba5752c52)

**Comment by MAKOMO** [comment]
Any reason why you have set the Host IP address in the Artisan WebSocket tab to `198.168.178.25` while the Probat machine indicates that it is using the `192.168.178.25`?

What happens if you fix that typo in the Artisan WebSocket tab and uses the correct IP address of the roasting machine?

Anyhow, it would be worse asking Probat if that machine also has a DNS name that could be used instead of the IP. The advantage would be that even if the IP address assigned to the Probat roaster by your router changes over time it would always be available under the same IP name and thus the configuration in Artisan would not need to be changed. I am aware that at least the Probatone series II machines did have an IP name. Not sure about the Series III.

**Comment by MatteoNat** [comment]
Sorry just a small mistake trying different things. The correct one was used several times. 
Thank you. I already sked for dns name but I still didn't have an answer

**Comment by MatteoNat** [comment]
Am I Right to say that: 
If the roaster is able to communicate with the laptop using the Ethernet cable and using the IP address in the google address bar, the roaster should be able to communicate with Artisan?
Pure genuine question

**Reply by poundy** [reply]
yes. If you have the correct destination IP address in Artisan, and have the browser talking to the same IP address, then things should work correctly (as the networking pieces are all correctly configured to permit this). It would be unusual, but is possible, that the Probat roaster is not exposing the same capabilities over the IP address as Artisan expects. Please do a test in the browser to http://192.168.178.25:80/websocket to test the response.

**Reply by MatteoNat** [reply]
I confirm: using the browser I do have connection.
Thank you Poundy

**Comment by MAKOMO** [comment, marked as answer]
Could you try a small thing: downgrade to [Artisan v2.10.0](https://github.com/artisan-roaster-scope/artisan/releases/tag/v2.10.0) and retry.

Rational: in Artisan v2.10.2 we upgrade the WebSocket communication which is used to communicate with newer Probat machines to a different library. That library seems to be using compression of the data by default, which is also standard. The previous lib used did not support compression at all. Maybe this difference can explain your connection issue. 

I am now adding a flag to deactivate compression for WebSocket communication and some additional logging. Thus if v2.10.0 still does not connect for you we can look at the log data to see what exactly is going wrong.

Sorry for your troubles with Artisan!
Marko

**Reply by MatteoNat** [reply]
Hi Makomo
Thank you for this link. 
The Probat P12 III, using Artisan v2.10.0, is now working. 

Thank you very much for your help

No need to be sorry, you guys are doing amazing giving this tools and assistance to roasters. 

Wish you all the best 

Matteo

**Reply by MAKOMO** [reply]
Sorry again for causing you troubles. Thanks for this confirmation. We will add the fix to the next version (a pre-release is already available as [continuous build](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous)). Maybe you find the time to confirm that this one works for you as well. This is essentially the v2.10.4 with the fix for the Probat WebSocket communication and should run very stable.

**Reply by MAKOMO** [reply]
Btw. the IP addresses in your pictures are not sensitive as they are valid for your network only (or actually the same numbers do exist in many other networks) and are for no use for an attacker.

**Comment by MatteoNat** [comment]
Hi Makomo, I do confirm. Version 2.10.5 is giving the readings. 
Many Thanks again

**Reply by MAKOMO** [reply]
Excellent news! Thanks for testing again and this feedback. So we cracked this one. Wonderful!

The fix will be in the next official Artisan version. You can keep using that 2.10.5 version which should run very stable until the next release.

**Reply by MatteoNat** [reply]
Amazing job guys! An unbelievably fast!

**Comment by MatteoNat** [comment]
Hi MAKOMO how are you?

 Unfortunately I had to change laptop as it crashes too many times. We just installed a router with sim card. 

I'm facing the same problem of communication with artisan.

 It looks I'm not able to download the v2.10.0. 

I'll send you some shots. 

to be notice: now the p12 III  ethernet cable is not able to communicate with the new laptop even if plugged.
It would be really handy for me to connect the router. 

Current situation. I am able to use only the ethernet connection with the old laptop

Regards 

Matteo
![Setting ip](https://github.com/user-attachments/assets/170187a8-af0a-4f57-b7f4-d323d7f6e3af)
![Static ip](https://github.com/user-attachments/assets/4456895f-f314-47f6-8e23-949a281f18b6)

**Reply by MAKOMO** [reply]
Always install the latest Artisan version which is v3.1.

Networking issues are outside the scope of this forum (we have a hard time to get Artisan specific issues sorted out here due to a lack of resources) and not in my expertise (I have no networking background). Please consult a local networking expert!

One note: the network configuration in your screenshots do not make much sense as the machines IP address is equal to that of the networks gateway. Every node in an IP network needs its own unique IP address.

**Reply by poundy** [reply]
Based on your earlier pics, the roaster itself is 192.168.178.25, but you are firstly trying to set the IP address of your PC to the same address, plus you're trying to set the gateway address to the same IP address, which absolutely will cause the error message you see. You should use some other random numbers on this isolated network; for example you could choose 192.168.178.11 for your PC and 192.168.178.1 for your default gateway, and I suspect that would work.
